### PR TITLE
return 200 even if no update and add activity to scheduled call update

### PR DESCRIPTION
**Pull request by codiebeulaine**
@justinvdm ready for review

### CODE REVIEW

File: mentorship/mentor/tests/test_views.py [hunk 1]
```diff
@@ -1434,6 +1412,30 @@ def test_schedule_detail_put(self):
...
+        activity.save_revision().publish()
+        data = json.dumps({
+            'activity': activity.pk,
+            'call_time': new_call_time.strftime('%Y-%m-%dT%H:%M')})
+        response = self.client.put(url, data, 'application/json')
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue(ScheduledCall.objects.filter(
+            activity=activity.pk).exists())
+
+        # test that response is 2 even if no update
```
Comment (justinvdm) [inline mentorship/mentor/tests/test_views.py]: Should this say `200`?

File: mentorship/mentor/views.py [hunk 2]
```diff
@@ -311,23 +310,32 @@ def perform_update(self, serializer):
...
+        if 'activity' in data:
+            queryset = ScheduledCall.objects.filter(
+                caller=caller, call_time=data['call_time'],
+                activity=data['activity']).exists()
+        else:
+            queryset = ScheduledCall.objects.filter(
+                caller=caller, call_time=data['call_time']).exists()
+        if queryset:
+            return HttpResponse(
+                'There is already a call for this date or with this actvity.')
```
Comment (justinvdm) [inline mentorship/mentor/views.py]: I probably didn't explain this clearly. I'd like us to respond with `200` for this case also. To the client's point of view, their update request should succeed regardless of whether anything or nothing changed as a result of their request.

Comment (codiebeulaine) [inline mentorship/mentor/views.py]: @justinvdm it does return a 200 response

Comment (codiebeulaine) [inline mentorship/mentor/views.py]: @justinvdm made some changes now

File: mentorship/mentor/tests/test_views.py (hunk 1 above)
Comment (codiebeulaine) [inline mentorship/mentor/tests/test_views.py]: yep

File: mentorship/mentor/views.py (hunk 2 above)
Comment (justinvdm) [inline mentorship/mentor/views.py]: Ah, sorry, misread :) This still needs changing though: it should respond with the model's data (the same as a normal update). Thought that was implied by the `200` but I should've made that clearer, sorry.

Comment (codiebeulaine) [inline mentorship/mentor/views.py]: @justinvdm I took that out now, look good?

Comment (justinvdm) [inline mentorship/mentor/views.py]: yep :) thanks

**Comment by justinvdm** [comment]
👍 wtih

**Pull request merged**